app/models/proxy.rb

```diff
@@ -347,16 +348,24 @@ def deploy!
     deploy
   end
 
+  def deploy_mesh_or_apicast
```

Comment:
I'd do another complete object for this logic, and use it here.
That way we can isolate the "deploy" logic with the update_attributes one

Reply:
By object do you mean a new class? Where would it be located, rails-ly speaking?